src/app/[id]/page.tsx

```diff
@@ -1,30 +1,23 @@
-import ArticlePreviewSection from "@/Components/ArticlePreviewSection/ArticlePreviewSection";
-import { notFound } from "next/navigation";
 import styles from "@/app/page.module.scss";
 import CategoriesBar from "@/Components/CategoriesBar/CategoriesBar";
-import PopularTags from "@/Components/PopularTags/PopularTags";
+import PopularTagsWrapper from "@/Components/PopularTags/PopularTagsWrapper";
 import { CategoriesType } from "@/lib/types/categories";
 import { fetchCategories } from "@/lib/api/categoryFetcher";
+import ArticlePreviewSectionWrapper from "@/Components/ArticlePreviewSection/ArticlePreviewSectionWrapper";
+import { notFound } from "next/navigation";
+import { ParamsType } from "@/lib/types/paramsType";
 
-type ParamsType = {
-  params: Promise<{ id: string }>;
-};
-
-const DisplayPostsByCategory = async ({ params }: ParamsType) => {
-  const id: string = (await params).id;
-  if (!id) {
-    notFound();
-  }
+const DisplayPosts = async ({ params }: ParamsType) => {
   const categories: CategoriesType = await fetchCategories();
-
+  const id = (await params).id;
```

Comment:
I think you meant to use `[slug]` here instead of id

Reply:
I renamed slug to id.